`OptionRanged`[something]'s `is_none` and `is_some` methods take ownership of the value

The standard library's `Option`'s [`is_none`](https://doc.rust-lang.org/std/option/enum.Option.html#method.is_none) and [`is_some`](https://doc.rust-lang.org/std/option/enum.Option.html#method.is_some) take the value by reference. 

I encountered this when using `serde`:

```rust
#[derive(Serialize)]
struct Struct {
    #[serde(skip_serializing_if = "OptionRangedU16::is_none")]
    number: OptionRangedU16<0, 1000>,
}
```

with the full compilation error:
```
error[E0308]: mismatched types
    --> src/main.rs:5:10
     |
5    | #[derive(Serialize)]
     |          ^^^^^^^^^ expected `OptionRangedU16<_, _>`, found `&OptionRangedU16<0, 1000>`
6    | struct Struct {
7    |     #[serde(skip_serializing_if = "OptionRangedU16::is_none")]
     |                                   -------------------------- arguments to this function are incorrect
     |
     = note: expected struct `OptionRangedU16<_, _>`
             found reference `&OptionRangedU16<0, 1000>`
note: method defined here
    --> C:\Users\Jacob\scoop\persist\rustup\.cargo\registry\src\index.crates.io-6f17d22bba15001f\deranged-0.3.11\src\lib.rs:1389:1
     |
1389 | / impl_ranged! {
1390 | |     RangedU8 {
1391 | |         mod_name: ranged_u8
1392 | |         internal: u8
...    |
1473 | |     }
1474 | | }
     | |_^
     = note: this error originates in the derive macro `Serialize` which comes from the expansion of the macro `impl_ranged` (in Nightly builds, run with -Z macro-backtrace for more info)

For more information about this error, try `rustc --explain E0308`.
```

The relevant part of the macro is here:
https://github.com/jhpratt/deranged/blob/c25ef2cefe36d92ebf3de6df2faaf293881f7349/src/lib.rs#L958-L970

where changing the `self` parameter to `&self` would be trivial, but would be a **breaking change**. 

Working around this on my end without a change to this crate is easy enough.

Thanks for your work on `deranged`, it's a great crate!

--- a/deranged/src/lib.rs
+++ b/deranged/src/lib.rs
@@ -1001,14 +1001,14 @@ macro_rules! impl_ranged {
 
             /// Returns `true` if the value is the niche value.
             #[inline(always)]
-            pub const fn is_none(self) -> bool {
+            pub const fn is_none(&self) -> bool {
                 <$type<MIN, MAX> as $crate::traits::RangeIsValid>::ASSERT;
                 self.get().is_none()
             }
 
             /// Returns `true` if the value is not the niche value.
             #[inline(always)]
-            pub const fn is_some(self) -> bool {
+            pub const fn is_some(&self) -> bool {
                 <$type<MIN, MAX> as $crate::traits::RangeIsValid>::ASSERT;
                 self.get().is_some()
             }

--- a/deranged/src/tests.rs
+++ b/deranged/src/tests.rs
@@ -148,6 +148,18 @@ macro_rules! tests {
         )*}
 
         #[test]
+        fn is_some_by_ref() {$(
+            let value = $opt::<5, 10>::Some($t::<5, 10>::MAX);
+            assert!($opt::is_some(&value));
+        )*}
+
+        #[test]
+        fn is_none_by_ref() {$(
+            let value = $opt::<5, 10>::None;
+            assert!($opt::is_none(&value));
+        )*}
+
+        #[test]
         fn default() {$(
             assert_eq!($opt::<5, 10>::default(), $opt::<5, 10>::None);
         )*}

